src/connectors/store.py

```
@@ -133,4 +221,54 @@ def flush(reason="batch"):
+    # Obtener todos los documentos para BM25
+    all_docs = get_all_documents(vectorstore)
+    
+    if not all_docs:
+        return None, []
+    
+    # Cargar o construir índice BM25
+    if BM25Retriever.index_exists(BM25_PATH) and not BM25Retriever.needs_rebuild(BM25_PATH, len(all_docs)):
+        print(f"📂 BM25: Cargando índice desde {BM25_PATH}")
+        bm25_retriever = BM25Retriever.load_local(BM25_PATH, k=k)
+    else:
+        print(f"🔨 BM25: Construyendo índice...")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)